app/components/dossiers/notified_toggle_component.rb

```diff
@@ -0,0 +1,45 @@
+class Dossiers::NotifiedToggleComponent < ApplicationComponent
+  delegate :reverse_order, to: :procedure_presentation
+
+  def initialize(procedure:, procedure_presentation:)
+    @procedure = procedure
+    @procedure_presentation = procedure_presentation
+    @current_sort = procedure_presentation.sort
+  end
+
+  def opposite_order
+    procedure_presentation.opposite_order_for(current_table, current_column)
+  end
+
+  def active?
+    sorted_by_notifications? && order_asc?
+  end
+
+  def icon_class_name
+    active? ? 'fr-fi-checkbox' : 'fr-fi-checkbox-blank'
+  end
```

Comment:
il me semble que tu peux tout mettre dans `private` https://viewcomponent.org/viewcomponents-at-github.html#most-viewcomponent-instance-methods-can-be-private